### PR TITLE
[FEATURE]Added cancel button on search bar when platform is iOS

### PR DESCRIPTION
**Pull request by juanpetterson**
@RocketChat/ReactNative

Added cancel button on search bar when platform is iOS, based on the iOS design guidelines.

### CODE REVIEW

**Comment by juanpetterson** [comment]
Closing for now, git is conflicting my commits.

**Comment by diegolmello** [comment]
@juanpetterson Please stop opening and closing the same PR over and over. Thanks.

**Comment by juanpetterson** [comment]
I was having trouble with my fork on the others, sorry about that.

File: app/views/DirectoryView/index.js [hunk 1]
```diff
@@ -139,13 +140,21 @@ class DirectoryView extends React.Component {
...
+
 	renderHeader = () => {
 		const { type } = this.state;
 		return (
 			<React.Fragment>
 				<SearchBox
 					onChangeText={this.onSearchChangeText}
 					onSubmitEditing={this.search}
+					hasCancel={isIOS}
+					onCancelPress={() => this.onCancelPress()}
```
Comment (djorkaeffalexandre) [inline app/views/DirectoryView/index.js]: can use `onCancelPress={this.onCancelPress}`

File: app/views/NewMessageView.js [hunk 2]
```diff
@@ -119,9 +119,19 @@ class NewMessageView extends React.Component {
...
+		Keyboard.dismiss();
+	}
+
 	renderHeader = () => (
 		<View>
-			<SearchBox onChangeText={text => this.onSearchChangeText(text)} testID='new-message-view-search' />
+			<SearchBox
+				onChangeText={text => this.onSearchChangeText(text)}
+				hasCancel={isIOS}
+				onCancelPress={() => this.onCancelPress()}
```
Comment (djorkaeffalexandre) [inline app/views/NewMessageView.js]: same here.

File: app/views/NewMessageView.js [hunk 3]
```diff
@@ -119,9 +119,19 @@ class NewMessageView extends React.Component {
...
+	onCancelPress = () => {
+		this.search('');
+		Keyboard.dismiss();
+	}
+
 	renderHeader = () => (
 		<View>
-			<SearchBox onChangeText={text => this.onSearchChangeText(text)} testID='new-message-view-search' />
+			<SearchBox
+				onChangeText={text => this.onSearchChangeText(text)}
```
Comment (djorkaeffalexandre) [inline app/views/NewMessageView.js]: can use `this.onSearchChangeText`

File: app/views/RoomMembersView/index.js [hunk 4]
```diff
@@ -242,8 +245,18 @@ class RoomMembersView extends React.Component {
...
 
+	onCancelPress = () => {
+		this.setState({ filtering: false, membersFiltered: [] });
+		Keyboard.dismiss();
+	}
+
 	renderSearchBar = () => (
-		<SearchBox onChangeText={text => this.onSearchChangeText(text)} testID='room-members-view-search' />
+		<SearchBox
+			onChangeText={text => this.onSearchChangeText(text)}
```
Comment (djorkaeffalexandre) [inline app/views/RoomMembersView/index.js]: same here.

File: app/views/RoomMembersView/index.js [hunk 5]
```diff
@@ -242,8 +245,18 @@ class RoomMembersView extends React.Component {
...
+			hasCancel={isIOS}
+			onCancelPress={() => this.onCancelPress()}
```
Comment (djorkaeffalexandre) [inline app/views/RoomMembersView/index.js]: same here.

File: app/views/SelectedUsersView.js [hunk 6]
```diff
@@ -163,9 +163,19 @@ class SelectedUsersView extends React.Component {
...
+	onCancelPress = () => {
+		this.search('');
+		Keyboard.dismiss();
+	}
+
 	renderHeader = () => (
 		<View style={styles.header}>
-			<SearchBox onChangeText={text => this.onSearchChangeText(text)} testID='select-users-view-search' />
+			<SearchBox
+				onChangeText={text => this.onSearchChangeText(text)}
```
Comment (djorkaeffalexandre) [inline app/views/SelectedUsersView.js]: same here.

File: app/views/SelectedUsersView.js [hunk 7]
```diff
@@ -163,9 +163,19 @@ class SelectedUsersView extends React.Component {
...
+				hasCancel={isIOS}
+				onCancelPress={() => this.onCancelPress()}
```
Comment (djorkaeffalexandre) [inline app/views/SelectedUsersView.js]: same here.

File: app/views/RoomMembersView/index.js (hunk 5 above)
Comment (juanpetterson) [reply app/views/RoomMembersView/index.js]: We need the onCancelPress function to dismiss  the keyboard, and change to only this.onSearchChangeText will throw an error. Here and in the others.

File: app/views/NewMessageView.js (hunk 3 above)
Comment (djorkaeffalexandre) [reply app/views/NewMessageView.js]: you can use only `this.onSearchChangeText` here

File: app/views/RoomMembersView/index.js (hunk 4 above)
Comment (djorkaeffalexandre) [reply app/views/RoomMembersView/index.js]: You can use only `this.onSearchChangeText` here.

File: app/views/SelectedUsersView.js (hunk 6 above)
Comment (djorkaeffalexandre) [reply app/views/SelectedUsersView.js]: same: https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/1136/files#r317420117

**Pull request closed without merging**